tests/test_kernels/lower_memory_free/src/bin/lower_memory_free.rs

```diff
@@ -0,0 +1,44 @@
+#![no_std] // don't link the Rust standard library
+#![no_main] // disable all Rust-level entry points
+
+use bootloader_api::{entry_point, info::MemoryRegionKind, BootInfo};
+use test_kernel_lower_memory_free::{exit_qemu, QemuExitCode};
+
+const LOWER_MEMORY_END_PAGE: u64 = 0x0010_0000;
+
+entry_point!(kernel_main);
+
+fn kernel_main(boot_info: &'static mut BootInfo) -> ! {
+    use core::fmt::Write;
+    use test_kernel_lower_memory_free::serial;
+
+    let mut count = 0;
+    for region in boot_info.memory_regions.iter() {
+        writeln!(
+            serial(),
+            "Region: {:016x}-{:016x} - {:?}",
+            region.start,
+            region.end,
+            region.kind
+        )
+        .unwrap();
+        if region.end <= LOWER_MEMORY_END_PAGE && region.kind == MemoryRegionKind::Usable {
+            let pages = (region.end - region.start) / 4096;
+            count += pages;
+        }
```

Comment:
We should also account for regions that don't end at `LOWER_MEMORY_END_PAGE`. For example:

```suggestion
        if region.kind == MemoryRegionKind::Usable && region.start < LOWER_MEMORY_END_PAGE {
            let end = core::cmp::min(region.end, LOWER_MEMORY_END_PAGE);
            let pages = (end - region.start) / 4096;
            count += pages;
        }
```